src/transactions/transactionBuilder/transactionBuilder.ts

```diff
@@ -234,7 +234,7 @@ export async function generateRawTransaction(args: {
   );
 
   const { maxGasAmount, gasUnitPrice, expireTimestamp } = {
-    maxGasAmount: BigInt(DEFAULT_MAX_GAS_AMOUNT),
+    maxGasAmount: options?.maxGasAmount ? BigInt(options.maxGasAmount) : BigInt(DEFAULT_MAX_GAS_AMOUNT),
     gasUnitPrice: BigInt(gasEstimate),
```

Comment:
Do we also care about amounts from gas unit price?

Reply:
we do https://github.com/aptos-labs/aptos-ts-sdk/pull/182/files#diff-97f79e1249bb5dfd1fc52b16da86fbcd1d517e8897c7cc0d70c79e73c61ecf3cR230